core/jobs/batch_jobs/suggestion_migration_jobs_test.py

```diff
@@ -0,0 +1,128 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for jobs.batch_jobs.suggestion_migration_jobs."""
+
+from __future__ import annotations
+
+from core import feconf
+from core.domain import exp_domain
+from core.domain import state_domain
+from core.jobs import job_test_utils
+from core.jobs.batch_jobs import suggestion_migration_jobs
+from core.jobs.types import job_run_result
+from core.platform import models
+
+MYPY = False
+if MYPY:
+    from mypy_imports import datastore_services
+
+(exp_models, suggestion_models) = models.Registry.import_models([
+    models.NAMES.exploration, models.NAMES.suggestion
+])
+datastore_services = models.Registry.import_datastore_services()
+
+
+class MigrateSuggestionJobTests(job_test_utils.JobTestBase):
+
+    JOB_CLASS = (
+        suggestion_migration_jobs
+        .RegenerateContentIdForTranslationSuggestionsInReviewJob
+    )
+    TARGET_ID = 'exp1'
+
+    def setUp(self):
+        super().setUp()
+        self.STATE_1 = state_domain.State.create_default_state(
+            feconf.DEFAULT_INIT_STATE_NAME, 'content_0', 'default_outcome_1',
+            is_initial_state=True).to_dict()
+        self.exp_1 = self.create_model(
+            exp_models.ExplorationModel,
+            id=self.TARGET_ID,
+            title='title',
+            init_state_name=feconf.DEFAULT_INIT_STATE_NAME,
+            category=feconf.DEFAULT_EXPLORATION_CATEGORY,
+            objective=feconf.DEFAULT_EXPLORATION_OBJECTIVE,
+            language_code='en',
+            tags=['Topic'],
+            blurb='blurb',
+            author_notes='author notes',
+            states_schema_version=feconf.CURRENT_STATE_SCHEMA_VERSION,
+            param_specs={},
+            param_changes=[],
+            auto_tts_enabled=feconf.DEFAULT_AUTO_TTS_ENABLED,
+            correctness_feedback_enabled=False,
+            states={feconf.DEFAULT_INIT_STATE_NAME: self.STATE_1}
```

Comment:
This state is already in the latest version, we are going to use the state in outdated schema (schema-50).

Basically, this one-off job is going to use exploration in old schema and a new-id-generator function to update the suggestion.

Reply:
Hm, would it be sufficient to set the schema version to the old version in line 63? Not sure.